realm/src/main/java/io/realm/RealmList.java

```diff
@@ -143,6 +181,18 @@ public E set(int location, E object) {
         return object;
     }
 
+    // Transparently copies a standalone object or managed object from another Realm to the Realm backing this RealmList.
+    private E copyToRealmIfNeeded(E object) {
+        if (object.row != null && object.realm.getPath().equals(realm.getPath())) {
```

Comment:
If we stored `path` as protected in Realm instances we could access this as a field instead of an accessor, which would be 7 times faster (according to Google)